shell/common/skia_event_tracer_impl.cc

```diff
@@ -183,22 +186,67 @@ class FlutterEventTracer : public SkEventTracer {
     trace_release_context(trace_context);
 
 #else   // defined(OS_FUCHSIA)
+    std::vector<const char*> arg_names;
```

Comment:
Can you either |reserve| or construct with a size of |num_args| here?  Since we use tracing to measure performance, it's very important that we minimize the performance impact of tracing, which means that we should avoid heap allocations in tracing internals whenever possible.

Reply:
Consider this an edit to above.  Now that I've had a little bit more time to think about it, could you use a |std::array| or C array here instead actually (so that we can skip the heap allocation entirely).

Reply:
Reserve the size to the 'num_args' .